### PR TITLE
PHP-1528: Ensure HAVE_MONGO_OPENSSL is defined on Windows

### PR DESCRIPTION
**Pull request by jmikola**
https://jira.mongodb.org/browse/PHP-1528

### CODE REVIEW

File: config.w32 [hunk 1]
```diff
@@ -19,6 +19,7 @@ if (PHP_MONGO != "no") {
   ADD_SOURCES(configure_module_dirname + "/batch", "delete.c insert.c update.c write.c", "mongo");
 
   AC_DEFINE('HAVE_MONGO', 1);
+  AC_DEFINE('HAVE_MONGO_OPENSSL', 1);
```
Comment (bjori) [inline config.w32]: 33      ADD_FLAG("CFLAGS_MONGO", " /D HAVE_MONGO_SSLLIB=1");

I'm curious, what is this?

Comment (jmikola) [reply config.w32]: I traced it back to https://github.com/mongodb/mongo-php-driver-legacy/commit/ec3b84c78f4dcb09d5b20b6d23736357daf5d305. Best I can tell, the constant is never used in the extension source.

Comment (jmikola) [reply config.w32]: Note: PHPC actually does defer its `AC_DEFINE()` for SSL to library detection (irrespective of depending on the `openssl` extension): https://github.com/mongodb/mongo-php-driver/blob/c3777b5a9d7286480739f8fe6680383c7c4abfff/config.w32#L156-L166

File: config.w32 [hunk 2]
```diff
@@ -29,7 +29,7 @@ if (PHP_MONGO != "no") {
   }
 
   if (CHECK_LIB("ssleay32.lib", "mongo", PHP_MONGO) && CHECK_LIB("libeay32.lib", "mongo", PHP_MONGO)) {
-    ADD_FLAG("CFLAGS_MONGO", " /D HAVE_MONGO_SSLLIB=1");
+    ADD_FLAG("CFLAGS_MONGO", " /D HAVE_MONGO_OPENSSL=1");
```
Comment (jmikola) [inline config.w32]: `HAVE_MONGO_SSLLIB` was originally added in https://github.com/mongodb/mongo-php-driver-legacy/commit/ec3b84c78f4dcb09d5b20b6d23736357daf5d305 but never used in the source. Changing this to conditionally define `HAVE_MONGO_OPENSSL` makes the Windows build consistent with `config.m4` calling `PHP_SETUP_OPENSSL()` (and is actually what `SETUP_OPENSSL()` does in newer PHP versions' `configure.js`).

Note that even if we don't find the libraries here, the driver can still establish SSL connections via PHP's streams API. That depends entirely on whether the OpenSSL extension is loaded. If we don't find the libraries here, that only means that `X509`, `GSSAPI` and `PLAIN` auth mechanisms will not be supported. In practice, PECL's build system should always find the libraries so even that case should never come up.

**Comment by bjori** [comment]
lgtm

**Pull request merged**